Controllers/ProjectControllers/Admin/UserController.php

```diff
@@ -0,0 +1,204 @@
+<?php
+declare(strict_types=1);
+
+namespace Controllers\ProjectControllers\Admin;
+
+use Models\ProjectModels\Admin\UserModel;
+use Controllers\ProjectControllers\BaseController;
+use Views\UserView;
+use Interfaces\IDataManagement;
+use Models\ProjectModels\DataRegistry;
+use Models\ProjectModels\Validation\Data\User\Admin\Validator;
+use Models\ProjectModels\Message\User\Admin\ResultMessageModel;
+
+/**
+ * @package Controllers\ProjectControllers\Admin
+ */
+class UserController extends BaseController
+{
+    private UserModel $userModel;
+
+    private UserView $userView;
+
+    private Validator $dataValidator;
+
+    private ResultMessageModel $msgModel;
+
+    /**
+     * Object for access to $_POST data
+     */
+    private IDataManagement $postInfo;
+
+    /**
+     * Object for access to $_FILE data
+     */
+    private IDataManagement $fileInfo;
+
+    public function __construct()
+    {
+        $this->userModel = new UserModel();
+        $this->userView = new UserView();
+        $this->postInfo = DataRegistry::getInstance()->get('post');
+
+        if ($this->postInfo->isPost()) {
+            $this->dataValidator = new Validator();
+            $this->msgModel = new ResultMessageModel();
+            $this->fileInfo = DataRegistry::getInstance()->get('file');
+        }
+    }
+
+    /**
+     * Render form for registration
+     *
+     * @param array|null $params
+     * @return void
+     */
+    public function registrationAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned()) {
+            $options = $this->userView->getOptions('Регистрация', 'admin/registration.phtml');
+            $this->userView->render($options);
+        } else {
+            $this->homeLocationByCustomerType();
+        }
+    }
+
+    /**
+     * Create new user
+     *
+     * @param array|null $params
+     * @return void
+     */
+    public function newAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned()) {
+            if ($this->postInfo->isPost()) {
+                $emptyResult = $this->dataValidator->emptyCheck($this->postInfo->getPostData(), 'registration');
+                if (!in_array(false, $emptyResult)) {
+                    $correctResult = $this->dataValidator->correctCheck($this->postInfo->getPostData());
+                    if (!in_array('', $correctResult)) {
+                        if ($this->fileInfo->isImageSent()) {
+                            if ($this->fileInfo->isImageType()) {
+                                if ($this->fileInfo->isSizeCorrect($params[0])) {
+                                    if (!$this->userModel->newUser($params[0], $correctResult)) {
```

Comment:
Eight if statements in a row within one method. Please think how to rework your method behaviour to avoid this fucking "ELKA"